main/define/src/mill/define/Task.scala

```diff
@@ -122,6 +106,18 @@ object Task extends TaskBase {
       cls: EnclosingClass
   ): Command[T] = macro Target.Internal.commandImpl[T]
 
+  def Command(
+               t: NamedParameterOnlyDummy.type = NamedParameterOnlyDummy,
+               exclusive: Boolean = false
+  ): CommandFactory = new CommandFactory(exclusive)
+  class CommandFactory private[mill] (val exclusive: Boolean) extends TaskBase.TraverseCtxHolder {
```

Comment:
why shouldn't this be a curried function?

Reply:
IIRC Scala 2 macros had an issue where they didnt allow you to call their parameters with explicit names. Actually I'm not sure if that still applies in 2.13.15 or 3.5.0, but it did at some point. Lemme try again to verify

Reply:
works in scala 3.5 anyway:
```scala
scala> def mcr[T: Type](fn: Expr[Either[String, T]], exclusive: Expr[Boolean])(using Quotes): Expr[Either[String, T]] = {
     |   import scala.concurrent.Future
     |   import scala.concurrent.ExecutionContext.Implicits.global
     |   import scala.concurrent.Await
     |   import scala.concurrent.duration.Duration
     |
     |   val exc = exclusive.valueOrAbort
     |   if exc then '{ println("EXCLUSIVE"); $fn }
     |   else '{ val fut = Future({Thread.sleep(1000); $fn}); Await.result(fut, Duration.Inf) }
     | }
def mcr
  [T]
    (fn: scala.quoted.Expr[Either[String, T]], exclusive:
      scala.quoted.Expr[Boolean])
      (implicit evidence$1: scala.quoted.Type[T], x$3: scala.quoted.Quotes):
        scala.quoted.Expr[Either[String, T]]

scala> {
     | inline def task[T](inline op: Either[String, T]) = ${ mcr[T]('op, 'false) }
     | inline def task[T](ignore: Int = -1, inline exclusive: Boolean = false)(inline op: Either[String, T]) = ${ mcr[T]('op, 'exclusive) }
     | }
def task[T](op: Either[String, T]): Either[String, T]
def task
  [T](ignore: Int, exclusive: Boolean)(op: Either[String, T]): Either[String, T]
```

Reply:
Yeah looks like it's still a limitation in 2.13.15

```scala
[14186] [error] /Users/lihaoyi/Github/mill/main/src/mill/main/MainModule.scala:403:17: macro applications do not support named and/or default arguments
[14186] [error]     Task.Command(exclusive = true) {
[14186] [error]                 ^
```

So likely we'll go with the boilerplate-y `CommandFactory` so this can land in 0.12.0, and in 0.13.0 we can convert to normal curried functions as one of the benefits of being on Scala 3.x